config/kernel.m4

```diff
@@ -275,6 +280,11 @@ AC_DEFUN([ZFS_AC_KERNEL_TEST_RESULT], [
 	ZFS_AC_KERNEL_IDMAP_MNT_API
 	ZFS_AC_KERNEL_IATTR_VFSID
 	ZFS_AC_KERNEL_FILEMAP
+	case "$host_cpu" in
+		powerpc*)
+			ZFS_AC_KERNEL_CPU_HAS_FEATURE
+			;;
+	esac
```

Comment:
Based on your nice analysis in this https://github.com/openzfs/zfs/issues/14545#issuecomment-1454695648 it seems like a slightly more concise way to handle this would be to make a small change like this to `include/os/linux/kernel/linux/simd_powerpc.h`.  This along with the configure check this should be enough.  No need for a new header or to include it in those other sources.  Would you mind testing this patch out on your system?  I just borrowed your comment and created a `#define` for `cpu_has_feature`  in the only place it should be needed.

```diff
diff --git a/include/os/linux/kernel/linux/simd_powerpc.h b/include/os/linux/kernel/linux/simd_powerpc.h
index 7faee70fe2..552e40e6f7 100644
--- a/include/os/linux/kernel/linux/simd_powerpc.h
+++ b/include/os/linux/kernel/linux/simd_powerpc.h
@@ -104,6 +104,17 @@
 #define        kfpu_init()             0
 #define        kfpu_fini()             ((void) 0)
 
+/*
+ * Linux 4.7 makes cpu_has_feature to use jump labels on powerpc if
+ * CONFIG_JUMP_LABEL_FEATURE_CHECKS is enabled, in this case however it
+ * references GPL-only symbol cpu_feature_keys. Therefore we overrides this
+ * interface when it is detected being GPL-only.
+ */
+#if defined(CONFIG_JUMP_LABEL_FEATURE_CHECKS) && \
+    defined(HAVE_CPU_HAS_FEATURE_GPL_ONLY)
+#define cpu_has_feature(feature)       early_cpu_has_feature(feature)
+#endif
+
 /*
  * Check if AltiVec instruction set is available
  */
```